### DISCUSSION TITLE
Tour steps that need to change app route

### DISCUSSION THREAD

**Discussion by dcworldwide** [gilbarbara/react-joyride#836] [Q&A]
I need to change the application route many times throughout my tour. This requires delaying the css targeting until the route is loaded. 

Generally speaking, I think this relates to a need to perform a async transition between steps...where some things must occur before the next step can run.

Is this possible?

**Comment by dcworldwide** [comment]
Based on the controlled callback example, I tried this... but it doesn't work... it create an infinite loop. 

I needed to run the route change, before the step rendered (before dom binding). But... step before can't be interrupted without the library bugging out. 

```
if (step?.route) {

                setState({
                    ...state,
                    running: false
                })

                wait(seconds(2)).then(() => {
                    emitRoute(step.route)
                    wait(seconds(2)).then(() => {
                        setState({
                            ...state,
                            running: true
                        })
                    })
                })
            }
```

**Comment by gilbarbara** [comment]
hey @dcworldwide 

Check the [MultiRoute example](https://github.com/gilbarbara/react-joyride-demo/blob/main/src/MultiRoute/Wrapper.tsx).
